pytorch_lightning/trainer/trainer.py

```diff
@@ -429,6 +429,13 @@ def __init__(
         """
         super().__init__()
         Trainer._log_api_event("init")
+
+        # warning for infinite epochs
+        if max_steps == -1 and max_epochs == -1:
+            warnings.warn(
+                "This configuration can result in infinite epochs, as a result of overriding training_epoch_end or validation_epoch_end with val_check_interval==float. This will keep the outputs in memory indefinitely."
+            )
```

Comment:
Hi @VedPatwardhan! Thank you for working on this!

Writing it here leads to raising the warning to other users who didn't override the hooks, so I think we should check if the hooks are overridden by user, too. You can see other warnings defined in configure_validator.py:

https://github.com/PyTorchLightning/pytorch-lightning/blob/5ad5ba54c0c477546d21daf75ac7b4748d7963a7/pytorch_lightning/trainer/configuration_validator.py#L243

Reply:
Got it. I'll make that change.

Reply:
Could you please check it now?

Reply:
@akihironitta could you please help me out understanding what's going wrong ?